sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/TablesExtensionConfigProvider.cs

```diff
@@ -245,51 +196,37 @@ public JObjectBuilder(TablesExtensionConfigProvider bindingProvider)
                 string finalQuery = attribute.Filter;
                 if (!string.IsNullOrEmpty(attribute.PartitionKey))
                 {
-                    var partitionKeyPredicate = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, attribute.PartitionKey);
+                    var partitionKeyPredicate = TableClient.CreateQueryFilter($"PartitionKey eq {attribute.PartitionKey}");
                     if (!string.IsNullOrEmpty(attribute.Filter))
                     {
-                        finalQuery = TableQuery.CombineFilters(attribute.Filter, TableOperators.And, partitionKeyPredicate);
+                        finalQuery = $"{partitionKeyPredicate} and {attribute.Filter}";
                     }
                     else
                     {
                         finalQuery = partitionKeyPredicate;
                     }
                 }
 
-                TableQuery tableQuery = new TableQuery
-                {
-                    FilterString = finalQuery
-                };
+                int? take = null;
                 if (attribute.Take > 0)
                 {
-                    tableQuery.TakeCount = attribute.Take;
+                    take = attribute.Take;
                 }
 
                 int countRemaining = attribute.Take;
 
                 JArray entityArray = new JArray();
-                TableContinuationToken token = null;
+                var entities = table.QueryAsync<TableEntity>(finalQuery, take, cancellationToken: cancellation).ConfigureAwait(false);
```

Comment:
since many of these params are strings, let's add parameter name hints here to be sure we're sending the correct ones.